### PR TITLE
Pcc02

### PR DESCRIPTION
**Pull request by fuzzygwalchmei**
ATTENTION: before clicking "Create Pull Request" please submit some meta data, thanks!

Difficulty level (1-10): [4]
Estimated time spent (hours): [2]
Completed (yes/no): [Yes]
I stretched my coding skills (if yes what did you learn?): [More list comprehensions, nesting functions, raising errors
GIT!!!!]
Other feedback (what can we improve?): []

### CODE REVIEW

**Comment by jcastle13** [comment]
Hello @fuzzygwalchmei 

The unit test pass but when using the code I noticed that the _validation function does not return a value just raise the ValueError exception.  If you can try to handle the _validation function by returning the word value.  I think besides that everything else looks good.

File: 02/fuzzygwalchmei/game.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+def get_possible_dict_words(draw):
+    """Get all possible words from draw which are valid dictionary words.
+    Use the _get_permutations_draw helper and DICTIONARY constant"""
+    return [word for word in _get_permutations_draw(draw) if word.lower() in DICTIONARY]
+
+
+def _get_permutations_draw(draw):
+    """Helper for get_possible_dict_words to get all permutations of draw letters.
+    Hint: use itertools.permutations"""
+    return ["".join(word) for words in [list(itertools.permutations(draw, n)) for n in range(1, NUM_LETTERS +1)] for word in words]
```
Comment (bbelderbos) [inline 02/fuzzygwalchmei/game.py:51]: break over multiple lines

File: 02/fuzzygwalchmei/game.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+
+# Below 2 functions pass through the same 'draw' argument (smell?).
+# Maybe you want to abstract this into a class?
+# get_possible_dict_words and _get_permutations_draw would be instance methods.
+# 'draw' would be set in the class constructor (__init__).
+def get_possible_dict_words(draw):
+    """Get all possible words from draw which are valid dictionary words.
+    Use the _get_permutations_draw helper and DICTIONARY constant"""
+    return [word for word in _get_permutations_draw(draw) if word.lower() in DICTIONARY]
```
Comment (bbelderbos) [inline 02/fuzzygwalchmei/game.py:45]: seems we do a `word in`, with a list it scans all items, would a `set` be better?

**Comment by bbelderbos** [comment]
Thanks @jcastle13, added a few more comments, happy to merge it in.

**Pull request merged**